Sources/Client.swift

```diff
@@ -277,6 +284,68 @@ extension Client {
     }
 
     /**
+     Fetch a collection of Assets from Contentful matching the specified query.
+
+     - Parameter query: The Query object to match results againts.
+     - Parameter completion: A handler being called on completion of the request.
+
+     - Returns: The data task being used, enables cancellation of requests.
+     */
+    @discardableResult public func fetchAssets(with query: AssetQuery,
+                                               then completion: @escaping ResultsHandler<ArrayResponse<Asset>>) -> URLSessionDataTask? {
+
+        let url = URL(forComponent: "assets", parameters: query.parameters)
+        return fetch(url: url, then: completion)
+    }
+
+    /**
+     Fetch a collection of Assets from Contentful matching the specified query.
+
+     - Parameter query: The Query object to match results againts.
+     - Returns: A tuple of data task and an observable for the resulting array of Assets.
+     */
+    @discardableResult public func fetchAssets(query: AssetQuery) -> Observable<Result<ArrayResponse<Asset>>> {
+        let asyncDataTask: AsyncDataTask<AssetQuery, ArrayResponse<Asset>> = fetchAssets(with:then:)
+        return toObservable(parameter: query, asyncDataTask: asyncDataTask).observable
+    }
+}
+
+
+// MARK: Mappable
```

Comment:
In swift, `// MARK:` adds annotations to make navigation in the project a bit easier. it should probably read `Modellable` instead of `Mappable` so i'll change that.

basic idea is to just help navigation while working.